src/citrine/resources/delete.py

```diff
@@ -71,31 +73,45 @@ def _async_gemd_batch_delete(
 
     if dataset_id is not None:
         body.update({'dataset_id': str(dataset_id)})
-
-    path = format_escaped_url('/projects/{project_id}/gemd/async-batch-delete',
-                              project_id=project_id
-                              )
+    if team_id is not None:
+        path = format_escaped_url('/teams/{team_id}/gemd/async-batch-delete',
+                                  team_id=team_id)
+    elif project_id is not None:
+        warn("project_id is deprecated as of 3.4.0, and will be removed in 4.0.0. Please provide "
+             "team_id instead",
+             DeprecationWarning)
+        path = format_escaped_url('/projects/{project_id}/gemd/async-batch-delete',
+                                  project_id=project_id)
+    else:
+        raise TypeError("Missing one required argument: team_id")
     response = session.post_resource(path, body)
 
     job_id = response["job_id"]
 
-    return _poll_for_async_batch_delete_result(project_id, session, job_id, timeout, polling_delay)
+    return _poll_for_async_batch_delete_result(
+        team_id=team_id,
+        session=session,
+        job_id=job_id,
+        project_id=project_id,
+        timeout=timeout,
+        polling_delay=polling_delay)
 
 
 def _poll_for_async_batch_delete_result(
-        project_id: UUID,
+        team_id: UUID,
         session: Session,
         job_id: str,
         timeout: float,
-        polling_delay: float
+        polling_delay: float,
+        project_id: UUID = None
```

Comment:
Similar for `_async_gemd_batch_delete`

Reply:
This might actually highlight the reason more clearly: despite it being preceded with an underscore, we actually import it (and `_async_gemd_batch_delete`) outside of their modules. Which we should definitely correct.

That being said, you are correct, as a result they don't require deprecation warnings.